FNP 2020 - FinCausal/baseline/task1/task1_baseline.py

```diff
@@ -0,0 +1,524 @@
+
+# Create a mask of 1s for each token followed by 0s for padding
+for seq in test_input_ids:
+  seq_mask = [float(i>0) for i in seq]
+  test_attention_masks.append(seq_mask) 
+
+# Convert to tensors.
+prediction_inputs = torch.tensor(test_input_ids).to(torch.int64)
+prediction_masks = torch.tensor(test_attention_masks)
+prediction_labels = torch.tensor(test_labels).to(torch.int64)
+
+# Set the batch size.  
+batch_size = 8  
+
+# Create the DataLoader.
+prediction_data = TensorDataset(prediction_inputs, prediction_masks, prediction_labels)
+prediction_sampler = SequentialSampler(prediction_data)
+prediction_dataloader = DataLoader(prediction_data, sampler=prediction_sampler, batch_size=batch_size)
+
+# Prediction on test set
+
+print('Predicting labels for {:,} test sentences...'.format(len(prediction_inputs)))
+
+# Put model in evaluation mode
+model.eval()
+
+# Tracking variables 
+predictions , true_labels = [], []
+
+# Predict 
+for batch in prediction_dataloader:
+  # Add batch to GPU
+  batch = tuple(t.to(device) for t in batch)
+  
+  # Unpack the inputs from our dataloader
+  b_input_ids, b_input_mask, b_labels = batch
+  
+  # Telling the model not to compute or store gradients, saving memory and 
+  # speeding up prediction
+  with torch.no_grad():
+      # Forward pass, calculate logit predictions
+      outputs = model(b_input_ids, token_type_ids=None, 
+                      attention_mask=b_input_mask)
+
+  logits = outputs[0]
+
+  # Move logits and labels to CPU
+  logits = logits.detach().cpu().numpy()
+  label_ids = b_labels.to('cpu').numpy()
+  
+  # Store predictions and true labels
+  predictions.append(logits)
+  true_labels.append(label_ids)
+
+print('    DONE.')
+
+print('Positive samples: %d of %d (%.2f%%)' % (test_df.label.sum(), len(test_df.label), (test_df.label.sum() / len(test_df.label) * 100.0)))
+
+matthews_set = []
+
+# Evaluate each test batch using Matthew's correlation coefficient
+print('Calculating Matthews Corr. Coef. for each batch...')
+
+# For each input batch...
+for i in range(len(true_labels)):
+  
+  # The predictions for this batch are a 2-column ndarray (one column for "0" 
+  # and one column for "1"). Pick the label with the highest value and turn this
+  # in to a list of 0s and 1s.
+  pred_labels_i = np.argmax(predictions[i], axis=1).flatten()
+  
+  # Calculate and store the coef for this batch.  
+  matthews = matthews_corrcoef(true_labels[i], pred_labels_i)                
+  matthews_set.append(matthews)
+  
+# Combine the predictions for each batch into a single list of 0s and 1s.
+flat_predictions = [item for sublist in predictions for item in sublist]
+flat_predictions = np.argmax(flat_predictions, axis=1).flatten()
+
+# Combine the correct labels for each batch into a single list.
+flat_true_labels = [item for sublist in true_labels for item in sublist]
+
+# Calculate the MCC
+mcc = matthews_corrcoef(flat_true_labels, flat_predictions)
+
+print('MCC: %.3f' % mcc)
+
```

Comment:
Run the suggestion before inserting to check the `flat_true_labels, flat_predictions` data type is accepted (should be ok as both are flat list but unran)

```suggestion

def evaluate(y_true, y_pred):
	"""
	Evaluate Precision, Recall, F1 scores between y_true and y_pred
	If output_file is provided, scores are saved in this file otherwise printed to std output.
	:param y_true: true labels
	:param y_pred: predicted labels
	:return: list of scores (F1, Recall, Precision, ExactMatch)
	"""
    
    # True Positive
    sub_tp = sum([1 for t, p in zip(y_true, y_pred) if p and p == t])
    # False Positive
    sub_fp = sum([1 for t, p in zip(y_true, y_pred) if p and p != t])
    # False Negative
    sub_fn = sum([1 for t, p in zip(y_true, y_pred) if p == 0 and p != t])

    precision = float(sub_tp) / float(sub_tp + sub_fp)
    recall = float(sub_tp) / float(sub_tp + sub_fn)
    f1 = (2.0 * precision * recall) / (precision + recall)

    scores = [
        "F1: %f\n" % f1,
        "Recall: %f\n" % recall,
        "Precision: %f\n" % precision,
        "ExactMatch: %f\n" % -1.0
    ]
    print(scores)

evaluate(flat_true_labels, flat_predictions)

```